MyOnboarding/src/MyOnboardingApp.Api/Controllers/TodoListController.cs

```diff
@@ -15,45 +15,163 @@ namespace MyOnboardingApp.Api.Controllers
     [Route("")]
     public class TodoListController : ApiController
     {
-        private readonly ITodoListRepository _repository;
+        private readonly IRetrieveItemService _retrieveService;
+        private readonly ICreateItemService _createService;
+        private readonly IUpdateItemService _editService;
+        private readonly IDeleteItemService _deleteService;
         private readonly IUrlLocator _urlLocator;
 
 
-        public TodoListController(ITodoListRepository repository, IUrlLocator urlLocator)
+        public TodoListController(IUrlLocator urlLocator, ICreateItemService createService, IUpdateItemService editService, 
+            IDeleteItemService deleteService, IRetrieveItemService retrieveService)
         {
-            _repository = repository;
+            _retrieveService = retrieveService;
+            _createService = createService;
+            _editService = editService;
+            _deleteService = deleteService;
             _urlLocator = urlLocator;
         }
 
 
         public async Task<IHttpActionResult> GetAsync() 
-            => Ok(await _repository.GetAllItemsAsync());
-
+            => Ok(await _retrieveService.GetAllItemsAsync());
+        
 
         [Route("{id}", Name = RoutesConfig.TodoListItemRouteName)]
-        public async Task<IHttpActionResult> GetAsync(Guid id) 
-            => Ok(await _repository.GetItemByIdAsync(id));
+        public async Task<IHttpActionResult> GetAsync(Guid id)
+        {
+            if (id == Guid.Empty)
+            {
+                ModelState.AddModelError("Id", "Id must not be empty.");
```

Comment:
You know you don't want your code to continue and that there is nothing to validate further. I dare to suggest:
```suggestion
                return BadRequest("Id must not be empty.");
```
Or if you insist on using model state then:
```suggestion
                ModelState.AddModelError(nameof(id), "Id must not be empty.");
                return BadRequest(ModelState);
```